apps/api/src/app/topics-v2/dtos/delete-topic-subscriptions.dto.ts

```diff
@@ -1,15 +1,36 @@
-import { ApiProperty } from '@nestjs/swagger';
-import { ArrayMaxSize, ArrayMinSize, IsArray, IsDefined } from 'class-validator';
+import { ApiExtraModels, ApiProperty, getSchemaPath } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import { ArrayMaxSize, ArrayMinSize, IsArray, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { TopicSubscriberIdentifierDto } from '../../shared/dtos/subscriptions/create-subscriptions.dto';
 
+@ApiExtraModels(TopicSubscriberIdentifierDto)
 export class DeleteTopicSubscriptionsRequestDto {
   @ApiProperty({
-    description: 'List of subscriber identifiers to unsubscribe from the topic (max: 100)',
+    description:
+      'List of subscriber identifiers to unsubscribe from the topic (max: 100). @deprecated Use the "subscriptions" property instead.',
     example: ['subscriberId1', 'subscriberId2'],
     type: [String],
+    deprecated: true,
   })
   @IsArray()
-  @IsDefined()
+  @IsString({ each: true })
+  @IsOptional()
   @ArrayMaxSize(100, { message: 'Cannot unsubscribe more than 100 subscribers at once' })
   @ArrayMinSize(1, { message: 'At least one subscriber identifier is required' })
-  subscriberIds: string[];
+  subscriberIds?: string[];
+
+  @ApiProperty({
+    description:
+      'List of subscriptions to unsubscribe from the topic (max: 100). Can be either a string array of subscriber IDs or an array of objects with identifier and subscriberId',
+    type: 'array',
+    items: {
+      oneOf: [{ type: 'string' }, { $ref: getSchemaPath(TopicSubscriberIdentifierDto) }],
+    },
+    example: [{ identifier: 'subscriber-123-subscription-a' }, { identifier: 'subscriber-456-subscription-b' }],
+  })
+  @IsArray()
+  @IsOptional()
+  @ArrayMaxSize(100, { message: 'Cannot unsubscribe more than 100 subscriptions at once' })
+  @ArrayMinSize(1, { message: 'At least one subscription is required' })
+  subscriptions?: Array<string | Pick<TopicSubscriberIdentifierDto, 'identifier'>>;
```

Comment:
Right now we only allow { identifier: string }, which lets us precisely delete a single subscription within a topic meaning only one specific resource is removed.

I’m wondering whether we should also allow passing a subscriberId, so the interface becomes { identifier: string, subscriberId: string }.
If we support subscriberId, the user would be able to delete all subscriptions within a topic belonging to that subscriber (in cases where they have more than one).